### PR TITLE
REST API: add `featured_media` support for video/audio

### PR DESCRIPTION
**Pull request by swissspidy**
Audio and Video attachments can have a featured image. This data is not currently returned by the `wp/v2/media` endpoint. This PR fixes that.

Supersedes #4645 and #4682 with a refreshed patch that incorporates all the feedback from the previous patches.

Trac ticket: https://core.trac.wordpress.org/ticket/41692

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php [hunk 1]
```diff
@@ -322,6 +330,42 @@ protected function insert_attachment( $request ) {
...
+	 *
+	 * @param int $featured_media Featured Media ID.
+	 * @param int $post_id        Post ID.
+	 * @return bool|WP_Error Whether the post thumbnail was successfully deleted, otherwise WP_Error.
+	 */
+	protected function handle_featured_media( $featured_media, $post_id ) {
+		$post_type         = get_post_type( $post_id );
+		$thumbnail_support = current_theme_supports( 'post-thumbnails', $post_type ) && post_type_supports( $post_type, 'thumbnail' ) || 'revision' === $post_type;
+
+		if ( ! $thumbnail_support && 'attachment' === $post_type && get_post_mime_type( $post_id ) ) {
```
Comment (TimothyBJacobs) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php]: Won't the post type always be attachment?

Comment (swissspidy) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php]: Hmm I suppose it is, perhaps unless someone is extending this controller for their post type or so.

Comment (swissspidy) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php]: Removed now 👍

**Review by TimothyBJacobs — APPROVED** [review]

**Comment by swissspidy** [comment]
Committed in https://core.trac.wordpress.org/changeset/57603

**Pull request closed without merging**